### DISCUSSION TITLE
use musescore with jack

### DISCUSSION THREAD

**Discussion by lyrra** [musescore/MuseScore#13745] [General]
Perhaps food for the handbook, how to connect musescore to jack.

Musescore can be bridged to jack by using a jack-alsa-plugin.

setup:

1) install jack2 and plugin that will add an frontend alsa api: musescore --> alsa-emulation --> jack --> real_alsa_hardware
```
# apt install jack2 libasound2-plugins
```
3) Get your jack devices:
```
$ jack_lsp
system:capture_1
system:capture_2
system:playback_1
system:playback_2
```
4) enter the devices into your asounrc dotfile:
```
$ cat > ~/.asoundrc
pcm.!default {
    type plug
    slave { pcm "jack" }
}

pcm.jack {
    type jack
    playback_ports {
        0 system:playback_1
        1 system:playback_2
    }
    capture_ports {
        0 system:capture_1
        1 system:capture_2
    }
}

ctl.mixer0 {
    type hw
    card 0
}
```
5) start jack:
```
jackd -r -d alsa
```

6) start musescore
  musescore should be able to connect to the alsa-jack-plugin.

**Comment by MarcSabatella** [comment]
Are you suggesting this works with MU4?

**Comment by lyrra** [comment]
Yes I'm using it with MU4.

**Comment by MarcSabatella** [comment]
Wow, that's great, will make people relying on this in MU3 happy!

**Comment by ariutti** [comment]
Is there a way to also have MuseScore4 output each instrument on its own stereo track?
I mean, get a multichannel output from MuseScore4 in order to be routed (via Jack) and later processed inside a DAW?

**Comment by diedeno** [comment]
This is no valid workaround. Come on. Why was jack developed in the first place it were this simple?

**Reply by cfirwin3** [reply]
Correct, there is no valid workaround.  But Jack was developed as a means to integrate multiple programs in a session for both audio AND midi  (which has implications for automation and synchronization through slave/master relationships).  If we are only talking about audio... then yes, Jack doesn't serve much of a purpose now, with a variety of session managers that can route basic raw audio sockets.  But all of the premium synchronization features that we had in M3 are now lost.  I propose just opening up the midi output sockets (including the real-time timecode) so that any midi timecode aware programs can at least sync to M4 as the master transport.  A lot of folks want this... but it would appear that we must all be users and not developers.

I would also reiterate that Jack is not necessarily antiquated or down and out.  Pipewire is the new audio engine standard on Linux and it emulates Jack so that Jack applications can function more seamlessly AND painlessly without the need for tweaking settings, etc.  All of the serious DAWS on Linux operate best (or exclusively) on Jack, so it is still VERY MUCH a vital platform for Linux audio, with no plans for that to change.

**Comment by cfirwin3** [comment]
To the film composers among us that are at a loss for options:
The loss of Jack sync tools has created a problem for those of us that were scoring to picture in previous versions of Musescore. Obviously, we can continue to use version 3.6 and export files to M4 for render, but this solution comes with zero promise for the future. So here is my sort of "survivalist guide" to composing to picture without any of the tools that have now become standard on all other serious notation programs. if you have never done this sort of work before and can't find resources on it (because the concept has long been antiquated), this "guide" goes out to you! It is a somewhat lengthy watch with some technical discussion... so grab a snack or a cup of coffee. The working document with the information discussed is available for download in the description.

https://youtu.be/FOwj4IlTZ-o

Take care!

-Carl

**Comment by rrohm** [comment]
Sorry, I cannot get the proposed workaround to work, i.e., it does not change anything.

**Reply by cfirwin3** [reply]
There is no workaround.  The initial post was for an early build and only allowed for audio output.  Jack is a real time sync platform and there has never been a capability to connect all of Jack to M4.  I see zero concern for reviving the capability... just a very outdated empty promise with no follow up.

Shame.
